app/db/queries/application.py

```diff
@@ -199,3 +199,286 @@ def clone_single_round(round_id, new_fund_id, new_short_name) -> Round:
+        new_component_config (dict): A dictionary containing the configuration for the new component.
+            new_component_config keys:
+                - page_id (str): The ID of the page to which the component belongs.
+                - theme_id (str): The ID of the theme to which the component belongs.
+                - title (str): The title of the component.
+                - hint_text (str): The hint text for the component.
+                - options (dict): The options such as classes, prefix etc
+                - type (str): The type of the component.
+                - template_name (str): The name of the template.
+                - is_template (bool): A flag indicating whether the component is a template.
+                - source_template_id (str): The ID of the source template.
+                - audit_info (dict): Audit information for the component.
+                - page_index (int): The index of the component within the page.
+                - theme_index (int): The index of the component within the theme.
+                - conditions (dict): The conditions such as potential routes based on the components value (can specify page path).
+                - runner_component_name (str): The name of the runner component.
+                - list_id (str): The ID of the list to which the component belongs.
+            Returns:
+                Component: The newly created component object.
+    """
+    # Instantiate the Component object with the provided and default values
+    component = Component(
+        component_id=uuid4(),
+        page_id=new_component_config.get("page_id", None),
+        theme_id=new_component_config.get("theme_id", None),
+        title=new_component_config.get("title"),
+        hint_text=new_component_config.get("hint_text"),
+        options=new_component_config.get("options", {}),
+        type=new_component_config.get("type"),   
+        is_template=new_component_config.get("is_template", False),
+        template_name=new_component_config.get("template_name", None),
+        source_template_id=new_component_config.get("source_template_id", None),
+        audit_info=new_component_config.get("audit_info", {}),
+        page_index=new_component_config.get("page_index"),
+        theme_index=new_component_config.get("theme_index"),
+        conditions=new_component_config.get("conditions", []),
+        runner_component_name=new_component_config.get("runner_component_name"),
+        list_id=new_component_config.get("list_id", None),
+    )
+
+    # Add the component to the session and commit
+    db.session.add(component)
+    db.session.commit()
+
+    # Return the created component object or its ID based on your requirements
+    return component
+
+def update_component(component_id, new_component_config):
+    component = db.session.query(Component).where(Component.component_id == component_id).one_or_none()
+    if component:
+        if "page_id" in new_component_config:
```

Comment:
in python cant we use a mapper to map these things ?

Reply:
good shout, i have pushed this :)